package/MDAnalysis/analysis/hbonds/wbridge_analysis.py

```diff
@@ -173,132 +166,316 @@ class WaterBridgeAnalysis_OtherFF(WaterBridgeAnalysis):
+            is the number of this type of water bridge.
+            The output of this frame is the sum of all the values in this dictionary.
+      '''
+
+      # decompose the first hydrogen bond.
+      s1_index, to_index, (s1_resname, s1_resid, s1_name),
+      (to_resname, to_resid, to_name), dist, angle = current[0]
+      # decompose the last hydrogen bond.
+      from_index, s2_index, (from_resname, from_resid, from_name),
+      (s2_resname, s2_resid, s2_name), dist, angle = current[-1]
+
+      # only the residue name is ASP and the atom name is OD1,
+      if s2_resname == 'ASP' and s2_name == 'OD1':
+          # only if the order of water bridge is less than 2
+          if len(current) -1 < 2:
+              # extract all water molecules involved in the water bridge
+              # extract the first water from selection 1
+              s1_index, to_index, (s1_resname, s1_resid, s1_name),
+              (to_resname, to_resid, to_name), dist, angle = current[0]
+              key = (to_resname, to_resid)
+              output[key] = 1
+
+              # extract all the waters between selection 1 and selection 2
+              for hbond in current[1:-1]:
+                  # decompose the hydrogen bond.
+                  from_index, to_index, (from_resname, from_resid, from_name),
+                  (to_resname, to_resid, to_name), dist, angle = hbond
+                  # add first water
+                  key1 = (from_resname, from_resid)
+                  output[key1] = 1
+                  # add second water
+                  key2 = (to_resname, to_resid)
+                  output[key2] = 1
+
+              # extract the last water to selection 2
+              from_index, s2_index, (from_resname, from_resid, from_name),
+              (s2_resname, s2_resid, s2_name), dist, angle = current[-1]
+              key = (from_resname, from_resid)
+              output[key] = 1
+
+  w.count_by_time(analysis_func=analysis)
+
+Returns ::
+
+  [1, 0,]
+
+Classes
+-------
+
+.. autoclass:: WaterBridgeAnalysis
+   :members:
+
+   .. attribute:: timesteps
+
+      List of the times of each timestep. This can be used together with
+      :attr:`~WaterBridgeAnalysis.timeseries` to find the specific time point
+      of a water bridge existence.
 
-         import pandas as pd
-         df = pd.DataFrame.from_records(w.table)
 """
-from __future__ import absolute_import, division
-import six
 
 from collections import defaultdict
 import logging
 import warnings
 
 import numpy as np
 
-from .hbond_analysis import HydrogenBondAnalysis
+from ..base import AnalysisBase
 from MDAnalysis.lib.NeighborSearch import AtomNeighborSearch
-from MDAnalysis.lib.log import ProgressMeter
+from MDAnalysis import NoDataError
 from MDAnalysis.lib import distances
-from MDAnalysis import SelectionWarning
 
 logger = logging.getLogger('MDAnalysis.analysis.wbridges')
 
-class WaterBridgeAnalysis(HydrogenBondAnalysis):
+class WaterBridgeAnalysis(AnalysisBase):
```

Comment:
Wouldn't it be better to transition `HydrogenBondAnalysis` to new-style analysis and then derive from there as before?

Reply:
@orbeckst Thank you very much for the advice. I wonder if I can have some guidance regarding how to make this pull request a better one, please? Thank you.

Reply:
If you can somehow unify the code base for the two analyses then that would be great. If it works better to derive from WaterBridgeAnalysis then that's fine with me.

However, breaking existing code (e.g. when changing how data are stored) is always critical. We can do it if really necessary but we really don't like to annoy users needlessly. 

Perhaps finish this PR first (make tests pass...) and then think about how to unify the two analysis classes. Many smaller PRs is better than one big one.

Reply:
@orbeckst Thank you for the comment. I have run the test myself and I'm not sure why the test isn't passing. I have borrowed my college's desktop and the tests have finished normally. I have open the log file in travis and it isn't showing any error.